ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-pim-sparse-mode.j2

```diff
@@ -5,9 +5,13 @@ router pim sparse-mode
 {%     if router_pim_sparse_mode.ipv4 is arista.avd.defined %}
    ipv4
 {%         for rp_address in router_pim_sparse_mode.ipv4.rp_addresses | arista.avd.natural_sort %}
-{%             for group in router_pim_sparse_mode.ipv4.rp_addresses[rp_address].groups | arista.avd.natural_sort %}
+{%             if router_pim_sparse_mode.ipv4.rp_addresses[rp_address].groups is arista.avd.defined and router_pim_sparse_mode.ipv4.rp_addresses[rp_address].groups | length > 0 %}
```

Comment:
The avd.defined test is enough to test. I don't see any essence of ```router_pim_sparse_mode.ipv4.rp_addresses[rp_address].groups | length > 0```, because arista.avd.defined will check both defined and not none.

Reply:
Same for below line 35.

Reply:
> The avd.defined test is enough to test. The condition, `router_pim_sparse_mode.ipv4.rp_addresses[rp_address].groups | length > 0` looks redundant, because arista.avd.defined will check both defined and not none.

Removed unnecessary check